docs/YIP/YIP-0012-consistent-price-verification.md

```diff
@@ -0,0 +1,39 @@
+# YIP-0012: Consistent Asset Price Verification for Margin Calls
+
+## Status
+
+Draft
+
+## Context
+
+In the context of trading platforms, specifically when handling margin calls, a significant challenge arises due to the volatility of asset price indices. Even if two brokers employ the same price index formula, discrepancies can occur when one broker sends a margin call to another. This is primarily because the asset price index can change rapidly, introducing potential inconsistencies in the pricing information.
+
+Using a fixed price range for accepting offers is not a foolproof solution, as it may not always be effective and can introduce a vulnerability to attacks.
+
+## Decision
+
+To address the challenges associated with margin calls and ensure consistency in asset pricing, we propose the following solution:
+
+### Price INDEX Definition
+
+- **Definition:** Each broker must adhere to a common definition of the price index. This includes using the same price feeds, weights, and formula for calculating the index.
+
+### Price History Buffer
+
+- **Buffer Implementation:** Each broker node will maintain a buffer of price history for a specified time range (e.g., the last 2 seconds). This buffer stores historical prices, providing a reference for verifying the accuracy of received margin calls. The timestamp of prices must be provided by the price sources to ensure every broker have the same data. 
```

Comment:
When we watched all the trades ClearPorts were overloaded. In this approach do we risk overloading?

Reply:
Also, do I understand correctly that each margin call will have prices from different sources in a range of 2 min? In this case, its quite a range of prices to collect, store, share and compare.

Or it will be only the calculated index prices calculated in the 2 minutes? In this variant, I believe the time of price index calculation can vary. Will we trust the latest timestamp out of two or there will be a different approach?